source/parse.h

```diff
@@ -3318,6 +3322,24 @@ class parser
                 }
             }
 
+            if (deduced_type) {
+                auto& type = std::get<declaration_node::object>(n->type);
+                if (peek(1)->type() == lexeme::Ampersand) {
+                    type->address_of = &curr();
+                }
+                else if (peek(1)->type() == lexeme::Multiply) {
+                    type->dereference_of = &curr();
+                    for (int i = 1; peek(i)->type() == lexeme::Multiply; ++i)
+                        type->dereference_cnt += 1;
+                }
+                else if (
+                    (peek(1)->type() == lexeme::LeftParen && curr().type() != lexeme::Colon)
```

Comment:
This is trying to avoid unnamed function expressions?

This makes me wonder whether this is happening too early... should we be doing this later, in `sema.h`, not during parsing? In sema we know whether it's an unnamed function declaration, we don't have to guess with a heuristic that can break.

Reply:
I will check if I can move that to sema.

Reply:
On second thought this heuristic is probably fine for this point-in-time localized feature.